backend/api/services.py

```diff
@@ -215,3 +221,172 @@ def get_reddit_posts(subreddit, symb, time="week") -> pd.DataFrame:
     df.replace("", np.nan, inplace=True)
     df.dropna(inplace=True)
     return df
+
+
+# TWEETS
+def get_tweets(query) -> pd.DataFrame:
+    """Returns a dataframe of tweets matching a given search query.
+
+    Args:
+        query (str): Search query eg. AAPL
+
+    Returns:
+        pd.DataFrame: DataFrame of tweets and other relevant information.
+    """
+    query = f"\\${query} lang:en"
+    tweets = twitter.search_recent_tweets(
+        query=query,
+        tweet_fields=["author_id", "created_at", "public_metrics", "entities"],
+        max_results=100,
+    )
+
+    tweets_df = pd.DataFrame(tweets.data)
+
+    # explode out public metrics
+    pub_metrics = tweets_df["public_metrics"].apply(pd.Series)
+
+    # extract hashtags
+    entities = pd.json_normalize(tweets_df["entities"])
+    hashtags = []
+    for i in entities["hashtags"]:
+        hashtags_str = ""
+        if isinstance(i, list):
+            for j in i:
+                hashtags_str += j["tag"] + " "
+        hashtags.append(hashtags_str)
```

Comment:
I think writing it as 
```python3
for i in entities["hashtags"]:
    hashtags.append(" ".join([j["tag"] for j in i if isinstance(i, list)]))
hashtags = pd.Series(hashtags).rename("hashtags")
```

is more pythonic and succinct

Reply:
will look into this more. currently this results in a type error.